### ISSUE TITLE
Build errors on the 4.1.0 branch

### ISSUE THREAD

**Issue by badnbusy** [dermotduffy/frigate-hass-card#1010]
- [ X ] I updated to the latest version available
- [ X ] I cleared the cache of my browser

Release with the issue: 4.1.0

Last working release (if known): 4.0.0

Browser and Operating System: Debian 11 / Chrome

Description of problem:

I'm trying to build the new 4.1.0 branch in order to test the improvements to go2rtc support, but getting errors on the build that I can't figure out:

```
~/frigate-hass-card$ nvm use 18.15.0
Now using node v18.15.0 (npm v9.5.0)
```

```
~/frigate-hass-card$ yarn install
yarn install v1.22.19
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.14s.
```

all good... but the build fails...

```
~/frigate-hass-card$ yarn run build
yarn run v1.22.19
$ yarn run lint && yarn run rollup
$ eslint 'src/**/*.ts'
$ rollup -c

src/card.ts → dist...
[!] (plugin rpt2) Error: src/types.ts:189:3 - error TS2345: Argument of type 'ZodObject<{ confirmation: ZodOptional<ZodUnion<[ZodBoolean, ZodObject<{ text: ZodOptional<ZodString>; exemptions: ZodOptional<ZodArray<ZodObject<{ user: ZodString; }, "strip", ZodTypeAny, { ...; }, { ...; }>, "many">>; }, "strip", ZodTypeAny, { ...; }, { ...; }>]>>; action: ZodLiteral<...>; }, "passthrough", ZodType...' is not assignable to parameter of type 'ZodType<CustomActionConfig & ExtendedConfirmationRestrictionConfig, any, any>'.
  Types of property '_type' are incompatible.
    Type 'objectOutputType<{ confirmation: ZodOptional<ZodUnion<[ZodBoolean, ZodObject<{ text: ZodOptional<ZodString>; exemptions: ZodOptional<ZodArray<ZodObject<{ user: ZodString; }, "strip", ZodTypeAny, { ...; }, { ...; }>, "many">>; }, "strip", ZodTypeAny, { ...; }, { ...; }>]>>; action: ZodLiteral<...>; }, ZodTypeAny, "pa...' is not assignable to type 'CustomActionConfig & ExtendedConfirmationRestrictionConfig'.
      Type 'objectOutputType<{ confirmation: ZodOptional<ZodUnion<[ZodBoolean, ZodObject<{ text: ZodOptional<ZodString>; exemptions: ZodOptional<ZodArray<ZodObject<{ user: ZodString; }, "strip", ZodTypeAny, { ...; }, { ...; }>, "many">>; }, "strip", ZodTypeAny, { ...; }, { ...; }>]>>; action: ZodLiteral<...>; }, ZodTypeAny, "pa...' is not assignable to type 'CustomActionConfig'.
        Types of property 'confirmation' are incompatible.
          Type 'boolean | { text?: string | undefined; exemptions?: { user: string; }[] | undefined; } | undefined' is not assignable to type 'ConfirmationRestrictionConfig | undefined'.
            Type 'false' has no properties in common with type 'ConfirmationRestrictionConfig'.

189   actionBaseSchema
      ~~~~~~~~~~~~~~~~
190     .extend({
    ~~~~~~~~~~~~~
...
192     })
    ~~~~~~
193     .passthrough(),
    ~~~~~~~~~~~~~~~~~~
src/types.ts:204:63 - error TS2339: Property 'extend' does not exist on type 'ZodType<CustomActionConfig & ExtendedConfirmationRestrictionConfig, any, any>'.

204 const frigateCardCustomActionsBaseSchema = customActionSchema.extend({
                                                                  ~~~~~~

src/types.ts

    at error (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:198:30)
    at throwPluginError (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:21718:12)
    at Object.error (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:22672:20)
    at Object.error (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:21895:42)
    at RollupContext.error (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/context.ts:81:17)
    at /home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/print-diagnostics.ts:37:17
    at Array.forEach (<anonymous>)
    at printDiagnostics (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/print-diagnostics.ts:9:14)
    at typecheckFile (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/index.ts:66:3)
    at Object.transform (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/index.ts:236:5)

error Command failed with exit code 1.

```

```

```

Thanks for any help.

**Comment by felipecrs** [comment]
You need to use yarn v3. You can use Volta (https://volta.sh) to simplify this process if you want.

Alternatively, you can take the built card from the GitHub Actions artifacts.

**Comment by badnbusy** [comment]
Still no joy with yarn 3.5...


```
yarn -v
3.5.0

~/frigate-hass-card$ yarn install
➤ YN0000: ┌ Resolution step
➤ YN0002: │ @types/cssnano@npm:5.1.0 doesn't provide postcss (p0e599), requested by cssnano
➤ YN0002: │ rollup-plugin-git-info@npm:1.0.0 doesn't provide rollup (p4d46c), requested by @rollup/plugin-json
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: Done with warnings in 0s 508ms

~/frigate-hass-card$ yarn run build

src/card.ts → dist...
[!] (plugin rpt2) Error: src/types.ts:189:3 - error TS2345: Argument of type 'ZodObject<{ confirmation: ZodOptional<ZodUnion<[ZodBoolean, ZodObject<{ text: ZodOptional<ZodString>; exemptions: ZodOptional<ZodArray<ZodObject<{ user: ZodString; }, "strip", ZodTypeAny, { ...; }, { ...; }>, "many">>; }, "strip", ZodTypeAny, { ...; }, { ...; }>]>>; action: ZodLiteral<...>; }, "passthrough", ZodType...' is not assignable to parameter of type 'ZodType<CustomActionConfig & ExtendedConfirmationRestrictionConfig, any, any>'.
  Types of property '_type' are incompatible.
    Type 'objectOutputType<{ confirmation: ZodOptional<ZodUnion<[ZodBoolean, ZodObject<{ text: ZodOptional<ZodString>; exemptions: ZodOptional<ZodArray<ZodObject<{ user: ZodString; }, "strip", ZodTypeAny, { ...; }, { ...; }>, "many">>; }, "strip", ZodTypeAny, { ...; }, { ...; }>]>>; action: ZodLiteral<...>; }, ZodTypeAny, "pa...' is not assignable to type 'CustomActionConfig & ExtendedConfirmationRestrictionConfig'.
      Type 'objectOutputType<{ confirmation: ZodOptional<ZodUnion<[ZodBoolean, ZodObject<{ text: ZodOptional<ZodString>; exemptions: ZodOptional<ZodArray<ZodObject<{ user: ZodString; }, "strip", ZodTypeAny, { ...; }, { ...; }>, "many">>; }, "strip", ZodTypeAny, { ...; }, { ...; }>]>>; action: ZodLiteral<...>; }, ZodTypeAny, "pa...' is not assignable to type 'CustomActionConfig'.
        Types of property 'confirmation' are incompatible.
          Type 'boolean | { text?: string | undefined; exemptions?: { user: string; }[] | undefined; } | undefined' is not assignable to type 'ConfirmationRestrictionConfig | undefined'.
            Type 'false' has no properties in common with type 'ConfirmationRestrictionConfig'.

189   actionBaseSchema
      ~~~~~~~~~~~~~~~~
190     .extend({
    ~~~~~~~~~~~~~
...
192     })
    ~~~~~~
193     .passthrough(),
    ~~~~~~~~~~~~~~~~~~
src/types.ts:204:63 - error TS2339: Property 'extend' does not exist on type 'ZodType<CustomActionConfig & ExtendedConfirmationRestrictionConfig, any, any>'.

204 const frigateCardCustomActionsBaseSchema = customActionSchema.extend({
                                                                  ~~~~~~

src/types.ts

    at error (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:198:30)
    at throwPluginError (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:21718:12)
    at Object.error (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:22672:20)
    at Object.error (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:21895:42)
    at RollupContext.error (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/context.ts:81:17)
    at /home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/print-diagnostics.ts:37:17
    at Array.forEach (<anonymous>)
    at printDiagnostics (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/print-diagnostics.ts:9:14)
    at typecheckFile (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/index.ts:66:3)
    at Object.transform (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/index.ts:236:5)

```

Do you have any steps on your other solution "take the built card from the GitHub Actions artifacts" ?

**Comment by felipecrs** [comment]
https://user-images.githubusercontent.com/29582865/225977705-e29797e6-420f-4099-b9ee-125e3d476048.mp4

**Comment by badnbusy** [comment]
Thank you... presume I need to merge all files into 'frigate-hass-card.js' ?

**Comment by felipecrs** [comment]
No, let me complete my recording.

**Comment by felipecrs** [comment]
https://user-images.githubusercontent.com/29582865/225981094-f5ab517d-6805-4b89-b0b3-a765a5fb6073.mp4

**Comment by badnbusy** [comment]
Done everything and force a browser refresh and also used another browser, but it's throwing console errors when trying to add the card to a view:

```
card-846b619a.js:11 Uncaught (in promise) TypeError: this._providerRef.value?.mute is not a function
    at E.mute (live-1e665240.js:81:992)
    at Object.m [as mute] (title-control-3d4097d8.js:6:1238)
    at R.autoMute (title-control-3d4097d8.js:13:9486)
    at j.updated (live-1e665240.js:30:990)
    at j._$AE (card-846b619a.js:11:5135)
    at j.performUpdate (card-846b619a.js:11:4915)
    at j.scheduleUpdate (card-846b619a.js:11:4497)
    at j._$Ej (card-846b619a.js:11:4405)
mute @ live-1e665240.js:81
m @ title-control-3d4097d8.js:6
autoMute @ title-control-3d4097d8.js:13
updated @ live-1e665240.js:30
_$AE @ card-846b619a.js:11
performUpdate @ card-846b619a.js:11
scheduleUpdate @ card-846b619a.js:11
_$Ej @ card-846b619a.js:11
await in _$Ej (async)
requestUpdate @ card-846b619a.js:11
set @ card-846b619a.js:11
j @ card-846b619a.js:17
_$AI @ card-846b619a.js:17
p @ card-846b619a.js:17
$ @ card-846b619a.js:17
_$AI @ card-846b619a.js:17
p @ card-846b619a.js:17
$ @ card-846b619a.js:17
_$AI @ card-846b619a.js:17
(anonymous) @ card-846b619a.js:23
update @ card-846b619a.js:23
performUpdate @ card-846b619a.js:11
scheduleUpdate @ card-846b619a.js:11
_$Ej @ card-846b619a.js:11
await in _$Ej (async)
requestUpdate @ card-846b619a.js:11
set @ card-846b619a.js:11
_intersectionHandler @ live-1e665240.js:6
card-846b619a.js:75 Error: No visible cameras found, you must configure at least one non-hidden camera
    at y_.initializeCameras (card-846b619a.js:366:15772)
    at async ky._initializeCameras (card-846b619a.js:366:74963)
    at async py.initializeIfNecessary (card-846b619a.js:366:52957)
ff @ card-846b619a.js:75
_handleThrownError @ card-846b619a.js:366
_initializeCameras @ card-846b619a.js:366
await in _initializeCameras (async)
(anonymous) @ card-846b619a.js:366
initializeIfNecessary @ card-846b619a.js:366
(anonymous) @ card-846b619a.js:366
Promise.then (async)
_initializeMandatory @ card-846b619a.js:366
shouldUpdate @ card-846b619a.js:366
performUpdate @ card-846b619a.js:11
scheduleUpdate @ card-846b619a.js:11
_$Ej @ card-846b619a.js:11
await in _$Ej (async)
requestUpdate @ card-846b619a.js:11
set @ card-846b619a.js:11
setConfig @ card-846b619a.js:366
value @ 42180fd1.js:1
performUpdate @ styles.ts:353
scheduleUpdate @ styles.ts:353
_$Ej @ styles.ts:353
await in _$Ej (async)
requestUpdate @ styles.ts:353
set @ styles.ts:353
j @ styles.ts:353
_$AI @ styles.ts:353
p @ styles.ts:353
$ @ styles.ts:353
_$AI @ styles.ts:353
j @ styles.ts:353
update @ styles.ts:353
performUpdate @ styles.ts:353
scheduleUpdate @ styles.ts:353
_$Ej @ styles.ts:353
await in _$Ej (async)
requestUpdate @ styles.ts:353
set @ styles.ts:353
value @ ha-dialog.ts:144
handleEvent @ styles.ts:353
i @ typography.js:181
value @ 3ba43f03.js:1
value @ 3ba43f03.js:1
value @ 3ba43f03.js:1
(anonymous) @ 3ba43f03.js:1
l @ card-846b619a.js:1
_updateConfig @ editor-615f4f97.js:218
d @ editor-615f4f97.js:126
(anonymous) @ editor-615f4f97.js:165
handleEvent @ card-846b619a.js:17



card-846b619a.js:75 Error: Failed to receive response from Home Assistant for request
    at Hg (card-846b619a.js:144:10229)
    at async U_ (card-846b619a.js:366:26002)
    at async r (card-846b619a.js:366:43230)
    at async Promise.all (:12443/lovelace/index 0)
    at async ly.getEvents (card-846b619a.js:366:43436)
    at async o (card-846b619a.js:366:19503)
    at async Promise.all (:12443/lovelace/index 0)
    at async s (card-846b619a.js:366:19919)
    at async Promise.all (:12443/lovelace/index 0)
    at async y_._handleQuery (card-846b619a.js:366:20001) {request: {…}, response: {…}}
ff @ card-846b619a.js:75
I @ media-to-view-30f41432.js:39
await in I (async)
C @ media-to-view-30f41432.js:39
_fetchMedia @ surround-0f0425c3.js:1
willUpdate @ surround-0f0425c3.js:1
performUpdate @ card-846b619a.js:11
scheduleUpdate @ card-846b619a.js:11
_$Ej @ card-846b619a.js:11
await in _$Ej (async)
requestUpdate @ card-846b619a.js:11
u @ card-846b619a.js:11
z @ card-846b619a.js:11
pe @ card-846b619a.js:23
l @ surround-0f0425c3.js:1
E @ typography.js:181
t @ typography.js:181
v @ card-846b619a.js:17
$ @ card-846b619a.js:17
_$AI @ card-846b619a.js:17
p @ card-846b619a.js:17
$ @ card-846b619a.js:17
_$AI @ card-846b619a.js:17
(anonymous) @ card-846b619a.js:23
update @ card-846b619a.js:23
performUpdate @ card-846b619a.js:11
scheduleUpdate @ card-846b619a.js:11
_$Ej @ card-846b619a.js:11
await in _$Ej (async)
requestUpdate @ card-846b619a.js:11
u @ card-846b619a.js:11
z @ card-846b619a.js:11
pe @ card-846b619a.js:23
B @ live-1e665240.js:6
E @ typography.js:181
t @ typography.js:181
v @ card-846b619a.js:17
$ @ card-846b619a.js:17
_$AI @ card-846b619a.js:17
p @ card-846b619a.js:17
$ @ card-846b619a.js:17
_$AI @ card-846b619a.js:17
p @ card-846b619a.js:17
$ @ card-846b619a.js:17
_$AI @ card-846b619a.js:17
(anonymous) @ card-846b619a.js:23
update @ card-846b619a.js:23
performUpdate @ card-846b619a.js:11
scheduleUpdate @ card-846b619a.js:11
_$Ej @ card-846b619a.js:11
await in _$Ej (async)
requestUpdate @ card-846b619a.js:11
(anonymous) @ card-846b619a.js:366
Promise.then (async)
_initializeMandatory @ card-846b619a.js:366
shouldUpdate @ card-846b619a.js:366
performUpdate @ card-846b619a.js:11
scheduleUpdate @ card-846b619a.js:11
_$Ej @ card-846b619a.js:11
await in _$Ej (async)
requestUpdate @ card-846b619a.js:11
set @ card-846b619a.js:11
setConfig @ card-846b619a.js:366
value @ 42180fd1.js:1
performUpdate @ styles.ts:353
scheduleUpdate @ styles.ts:353
_$Ej @ styles.ts:353
await in _$Ej (async)
requestUpdate @ styles.ts:353
set @ styles.ts:353
j @ styles.ts:353
_$AI @ styles.ts:353
p @ styles.ts:353
$ @ styles.ts:353
_$AI @ styles.ts:353
j @ styles.ts:353
update @ styles.ts:353
performUpdate @ styles.ts:353
scheduleUpdate @ styles.ts:353
_$Ej @ styles.ts:353
await in _$Ej (async)
requestUpdate @ styles.ts:353
set @ styles.ts:353
value @ ha-dialog.ts:144
handleEvent @ styles.ts:353
i @ typography.js:181
value @ 3ba43f03.js:1
value @ 3ba43f03.js:1
value @ 3ba43f03.js:1
(anonymous) @ 3ba43f03.js:1
l @ card-846b619a.js:1
_updateConfig @ editor-615f4f97.js:218
_valueChangedHandler @ editor-615f4f97.js:355
(anonymous) @ editor-615f4f97.js:21
handleEvent @ card-846b619a.js:17
i @ typography.js:181
(anonymous) @ b894386b.js:1
setTimeout (async)
value @ b894386b.js:1
value @ b894386b.js:1
handleEvent @ styles.ts:353
i @ typography.js:181
value @ 18580e4f.js:1
handleEvent @ styles.ts:353
O @ typography.js:181
D @ typography.js:181
P @ typography.js:181
(anonymous) @ typography.js:181
_propertiesChanged @ typography.js:181
_propertiesChanged @ vaadin-combo-box-scroller.js:50
_flushProperties @ typography.js:181
_invalidateProperties @ typography.js:181
Object.defineProperty.set @ typography.js:181
_selectedItemChanged @ vaadin-combo-box-scroller.js:50
U @ typography.js:181
C @ typography.js:181
_propertiesChanged @ typography.js:181
_propertiesChanged @ vaadin-combo-box-scroller.js:50
_flushProperties @ typography.js:181
_invalidateProperties @ typography.js:181
Object.defineProperty.set @ typography.js:181
_commitValue @ vaadin-combo-box-scroller.js:50
_onClosed @ vaadin-combo-box-scroller.js:50
_openedChanged @ vaadin-combo-box-scroller.js:50
A @ typography.js:181
C @ typography.js:181
_propertiesChanged @ typography.js:181
_propertiesChanged @ vaadin-combo-box-scroller.js:50
_flushProperties @ typography.js:181
_invalidateProperties @ typography.js:181
Object.defineProperty.set @ typography.js:181
close @ vaadin-combo-box-scroller.js:50
_overlaySelectedItemChanged @ vaadin-combo-box-scroller.js:50
__onItemClick @ vaadin-combo-box-scroller.js:50


card-846b619a.js:75 Error: Failed to receive response from Home Assistant for request
    at Hg (card-846b619a.js:144:10229)
    at async U_ (card-846b619a.js:366:26002)
    at async r (card-846b619a.js:366:43230)
    at async Promise.all (:12443/lovelace/index 0)
    at async ly.getEvents (card-846b619a.js:366:43436)
    at async o (card-846b619a.js:366:19503)
    at async Promise.all (:12443/lovelace/index 0)
    at async s (card-846b619a.js:366:19919)
    at async Promise.all (:12443/lovelace/index 0)
    at async y_._handleQuery (card-846b619a.js:366:20001) {request: {…}, response: {…}}
```

**Comment by dermotduffy** [comment]
@badnbusy

For the `v4.1.0` branch you must upgrade:

* Your Frigate integration to the latest beta: https://github.com/blakeblackshear/frigate-hass-integration/releases/tag/v4.0.0-beta4
* Your Frigate backend to the latest beta: https://github.com/blakeblackshear/frigate/releases/tag/v0.12.0-beta9

Based on that error, at least one of the above is likely untrue in your case (probably the integration).

**Comment by badnbusy** [comment]
> 

@dermotduffy  Thanks. So you were correct, I was only on v3.x of the hass-integration. I am already on beta9 of the Frigate backend. 

So I removed both the integration and card again, re-installed the integration at 4.0.0-beta4, and then re-added the integration and I'm still seeing errors and the card doesn't load any camera (forced a refresh and tried another browser)

![image](https://user-images.githubusercontent.com/11709354/226099057-b1d0a420-9592-41ff-9253-dc6548c9f755.png)

![image](https://user-images.githubusercontent.com/11709354/226099110-9e6dfae5-4bfc-4fa2-9754-e7ca45e0b76d.png)

Any thoughts?

Thanks

**Comment by dermotduffy** [comment]
@badnbusy Can you hold down the Frigate icon/button on the card for a few seconds then let go? It'll show you diagnostics information that contains the version of the integration it is communicating with (as I still think you are somehow still using an older version of the integration).

**Comment by badnbusy** [comment]
> 

Here's the card diagnostics:

```
Card diagnostics. Please review for confidential information prior to sharing
{
  "ha_version": "2023.3.3",
  "card_version": "4.0.0",
  "browser": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36",
  "date": "2023-03-18T19:27:14.708Z",
  "frigate_version": {
    "deac9914db67bc53bd5c9b105ddf8f94": "3.0.1/0.12.0-e8d8cc4"
  },
  "lang": "en",
  "timezone": "Europe/London",
  "git": {
    "build_version": "4.0.0-release-4.1.0+g5a48951",
    "build_date": "Tue, 14 Mar 2023 05:08:04 GMT",
    "commit_date": "Mon, 13 Mar 2023 22:06:29 -0700"
  },
  "config": {
    "type": "custom:frigate-card",
    "cameras": [
      {
        "camera_entity": "camera.front"
      },
      {
        "camera_entity": "camera.back"
      },
      {
        "camera_entity": "camera.garden"
      },
      {
        "camera_entity": "camera.side"
      },
      {
        "camera_entity": "camera.drive"
      },
      {
        "camera_entity": "camera.porch"
      },
      {
        "camera_entity": "camera.office"
      },
      {
        "camera_entity": "camera.office_inside"
      }
    ],
    "view": {
      "default": "live"
    },
    "menu": {
      "buttons": {
        "snapshots": {
          "enabled": false
        },
        "download": {
          "enabled": false
        },
        "frigate_ui": {
          "enabled": false
        },
        "timeline": {
          "enabled": false
        }
      }
    },
    "dimensions": {
      "aspect_ratio_mode": "dynamic"
    }
  }
}
```

**Comment by badnbusy** [comment]
It now works and is correctly using the latest v4 beta, thanks so much for your help.

Last question - I am using go2rtc in Frigate and have both sub and main streams. I have the card configured to use the go2rtc option and have set the stream name to the sub in order to improve the load time on the HA interface, but is there a way when double clicking/full-screen mode to then show the main stream so that it's better quality? Or another way to dynamically switch between stream names?

**Comment by felipecrs** [comment]
Yes, there is. Do a search for substream in the v4.1.0 README, there are some examples there.

**Comment by badnbusy** [comment]
I've tried this but it kills my HA dashboard that I use for camera's.

The goal is to have a substream feed for all eight cameras, and when clicking the HD button when viewing one, it loads the main stream. Here is the problem when I click the tab:

![image](https://user-images.githubusercontent.com/11709354/226434431-e536b69a-957b-40e5-b01d-2738e5ef09d2.png)

The lovelace and frigate card yaml:

```
  - theme: Backend-selected
    title: CAMERAS
    path: cameras
    badges: []
    cards:
      - type: custom:frigate-card
        cameras:
          - camera_entity: camera.front
            live_provider: go2rtc
            dependencies:
              cameras:
                - front_hd
            go2rtc:
              modes:
                - mse
              stream: front_sub
          - camera_entity: camera.front
            id: front_hd
            live_provider: go2rtc
            go2rtc:
              modes:
                - mse
            occupancy: false
            hide: true
          - camera_entity: camera.back
            live_provider: go2rtc
            dependencies:
              cameras:
                - back_hd
            go2rtc:
              modes:
                - mse
              stream: back_sub
          - camera_entity: camera.back
            id: back_hd
            live_provider: go2rtc
            go2rtc:
              modes:
                - mse
            occupancy: false
            hide: true
          - camera_entity: camera.garden
            live_provider: go2rtc
            dependencies:
              cameras:
                - garden_hd
            go2rtc:
              modes:
                - mse
              stream: garden_sub
          - camera_entity: camera.garden
            id: garden_hd
            live_provider: go2rtc
            go2rtc:
              modes:
                - mse
            occupancy: false
            hide: true
          - camera_entity: camera.side
            live_provider: go2rtc
            dependencies:
              cameras:
                - side_hd
            go2rtc:
              modes:
                - mse
              stream: side_sub
          - camera_entity: camera.side
            id: side_hd
            live_provider: go2rtc
            go2rtc:
              modes:
                - mse
            occupancy: false
            hide: true
          - camera_entity: camera.drive
            live_provider: go2rtc
            dependencies:
              cameras:
                - drive_hd
            go2rtc:
              modes:
                - mse
              stream: drive_sub
          - camera_entity: camera.drive
            id: drive_hd
            live_provider: go2rtc
            go2rtc:
              modes:
                - mse
            occupancy: false
            hide: true
          - camera_entity: camera.porch
            live_provider: go2rtc
            dependencies:
              cameras:
                - porch_hd
            go2rtc:
              modes:
                - mse
              stream: porch_sub
          - camera_entity: camera.porch
            id: porch_hd
            live_provider: go2rtc
            go2rtc:
              modes:
                - mse
            occupancy: false
            hide: true
          - camera_entity: camera.office
            live_provider: go2rtc
            dependencies:
              cameras:
                - office_hd
            go2rtc:
              modes:
                - mse
              stream: office_sub
          - camera_entity: camera.office
            id: office_hd
            live_provider: go2rtc
            go2rtc:
              modes:
                - mse
            occupancy: false
            hide: true
          - camera_entity: camera.office_inside
            live_provider: go2rtc
            dependencies:
              cameras:
                - office_inside_hd
            go2rtc:
              modes:
                - mse
              stream: office_inside_sub
          - camera_entity: camera.office_inside
            id: office_inside_hd
            live_provider: go2rtc
            go2rtc:
              modes:
                - mse
            occupancy: false
            hide: true
        menu:
          buttons:
            snapshots:
              enabled: false
            download:
              enabled: false
            timeline:
              enabled: false
            substreams:
              icon: mdi:high-definition
              alignment: opposing
              enabled: true
            camera_ui:
              enabled: false
            expand:
              enabled: false
            fullscreen:
              enabled: true
              alignment: opposing
          style: overlay
          position: top
        live:
          lazy_load: false
          draggable: false
        view:
          default: live
        performance:
          profile: low
          features:
            animated_progress_indicator: true
        dimensions:
          aspect_ratio_mode: dynamic
```

**Comment by felipecrs** [comment]
Maybe you are affected by:

- https://github.com/dermotduffy/frigate-hass-card/issues/1000

Try reverting the PR and building the card again to see if the issue remains...

```console
git revert 7eb130ea54987c9ad36f20f10006482571d1cf9a.
```

**Comment by felipecrs** [comment]
Interesting. I will try changing this on my side to see if it helps with https://github.com/dermotduffy/frigate-hass-card/issues/1000 as well.

**Comment by badnbusy** [comment]
Off-topic, where is the best place to ask general questions? I don't like raising new topics as that implies it's a "bug"...

i.e. is there any way to make the hass card bigger? I want it to fill the dashboard screen not just be a fixed square. I realise this isn't a hass card specific issue so just looking for pointers!

**Comment by badnbusy** [comment]
Hmm, this also kills the card in the same way, 10% cpu usage and renders the entire HA interface unusable:

```
menu:
          buttons:
            snapshots:
              enabled: false
            download:
              enabled: false
            timeline:
              enabled: false
            substreams:
              icon: mdi:high-definition
              alignment: opposing
              enabled: true
            camera_ui:
              enabled: false
            expand:
              enabled: false
            fullscreen:
              enabled: true
              alignment: opposing
            frigate:
              enabled: true
            **media_player:
              enabled: false**
```

Setting back to this works:

```
            **media_player:
              enabled: true**
```

**Comment by felipecrs** [comment]
> i.e. is there any way to make the hass card bigger? I want it to fill the dashboard screen not just be a fixed square. I realise this isn't a hass card specific issue so just looking for pointers!

Don't you mean a 1-card panel dashboard type?

**Comment by dermotduffy** [comment]
> Still no joy with yarn 3.5...
> 
> ```
> yarn -v
> 3.5.0
> 
> ~/frigate-hass-card$ yarn install
> ➤ YN0000: ┌ Resolution step
> ➤ YN0002: │ @types/cssnano@npm:5.1.0 doesn't provide postcss (p0e599), requested by cssnano
> ➤ YN0002: │ rollup-plugin-git-info@npm:1.0.0 doesn't provide rollup (p4d46c), requested by @rollup/plugin-json
> ➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
> ➤ YN0000: └ Completed
> ➤ YN0000: ┌ Fetch step
> ➤ YN0000: └ Completed
> ➤ YN0000: ┌ Link step
> ➤ YN0000: └ Completed
> ➤ YN0000: Done with warnings in 0s 508ms
> 
> ~/frigate-hass-card$ yarn run build
> 
> src/card.ts → dist...
> [!] (plugin rpt2) Error: src/types.ts:189:3 - error TS2345: Argument of type 'ZodObject<{ confirmation: ZodOptional<ZodUnion<[ZodBoolean, ZodObject<{ text: ZodOptional<ZodString>; exemptions: ZodOptional<ZodArray<ZodObject<{ user: ZodString; }, "strip", ZodTypeAny, { ...; }, { ...; }>, "many">>; }, "strip", ZodTypeAny, { ...; }, { ...; }>]>>; action: ZodLiteral<...>; }, "passthrough", ZodType...' is not assignable to parameter of type 'ZodType<CustomActionConfig & ExtendedConfirmationRestrictionConfig, any, any>'.
>   Types of property '_type' are incompatible.
>     Type 'objectOutputType<{ confirmation: ZodOptional<ZodUnion<[ZodBoolean, ZodObject<{ text: ZodOptional<ZodString>; exemptions: ZodOptional<ZodArray<ZodObject<{ user: ZodString; }, "strip", ZodTypeAny, { ...; }, { ...; }>, "many">>; }, "strip", ZodTypeAny, { ...; }, { ...; }>]>>; action: ZodLiteral<...>; }, ZodTypeAny, "pa...' is not assignable to type 'CustomActionConfig & ExtendedConfirmationRestrictionConfig'.
>       Type 'objectOutputType<{ confirmation: ZodOptional<ZodUnion<[ZodBoolean, ZodObject<{ text: ZodOptional<ZodString>; exemptions: ZodOptional<ZodArray<ZodObject<{ user: ZodString; }, "strip", ZodTypeAny, { ...; }, { ...; }>, "many">>; }, "strip", ZodTypeAny, { ...; }, { ...; }>]>>; action: ZodLiteral<...>; }, ZodTypeAny, "pa...' is not assignable to type 'CustomActionConfig'.
>         Types of property 'confirmation' are incompatible.
>           Type 'boolean | { text?: string | undefined; exemptions?: { user: string; }[] | undefined; } | undefined' is not assignable to type 'ConfirmationRestrictionConfig | undefined'.
>             Type 'false' has no properties in common with type 'ConfirmationRestrictionConfig'.
> 
> 189   actionBaseSchema
>       ~~~~~~~~~~~~~~~~
> 190     .extend({
>     ~~~~~~~~~~~~~
> ...
> 192     })
>     ~~~~~~
> 193     .passthrough(),
>     ~~~~~~~~~~~~~~~~~~
> src/types.ts:204:63 - error TS2339: Property 'extend' does not exist on type 'ZodType<CustomActionConfig & ExtendedConfirmationRestrictionConfig, any, any>'.
> 
> 204 const frigateCardCustomActionsBaseSchema = customActionSchema.extend({
>                                                                   ~~~~~~
> 
> src/types.ts
> 
>     at error (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:198:30)
>     at throwPluginError (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:21718:12)
>     at Object.error (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:22672:20)
>     at Object.error (/home/james/frigate-hass-card/node_modules/rollup/dist/shared/rollup.js:21895:42)
>     at RollupContext.error (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/context.ts:81:17)
>     at /home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/print-diagnostics.ts:37:17
>     at Array.forEach (<anonymous>)
>     at printDiagnostics (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/print-diagnostics.ts:9:14)
>     at typecheckFile (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/index.ts:66:3)
>     at Object.transform (/home/james/frigate-hass-card/node_modules/rollup-plugin-typescript2/src/index.ts:236:5)
> ```
> 
> Do you have any steps on your other solution "take the built card from the GitHub Actions artifacts"? I can see the artifact file but is it 4.1.0 or 4.0.0?

FYI: I encountered the same build issue and already fixed it, in the now re-numbered `v5.0.0` branch (previously named `v4.1.0`).

As such, I don't think there's anything left to do here. Please re-open if I'm mistaken.
